omod/src/main/java/org/openmrs/web/rest/resource/TagResource.java

```diff
@@ -0,0 +1,199 @@
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("uuid");
+		description.addProperty("tag");
+		description.addLink(delegate.getObjectType().substring(delegate.getObjectType().lastIndexOf(".") + 1).toLowerCase(),
+		    "/" + getUriOfObject(delegate.getObjectType(), delegate.getObjectUuid()));
+		description.addSelfLink();
+		description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		return convertDelegateToRepresentation(delegate, description);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#getRepresentationDescription(Representation)
+	 */
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = null;
+		if (rep instanceof RefRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("tag");
+			description.addProperty("objectType");
+			description.addProperty("objectUuid");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		} else if (rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("tag");
+			description.addProperty("objectType");
+			description.addProperty("objectUuid");
+			description.addProperty("auditInfo");
+			description.addSelfLink();
+		}
+		return description;
+	}
+	
+	/**
+	 * @return the TagService
+	 */
+	private TagService getService() {
+		return Context.getService(TagService.class);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#getCreatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		
+		description.addRequiredProperty("tag");
+		description.addRequiredProperty("objectType");
+		description.addRequiredProperty("objectUuid");
+		
+		return description;
+	}
+	
+	/**
+	 * @see DelegatingCrudResource#doGetAll(RequestContext)
+	 */
+	@Override
+	protected NeedsPaging<Tag> doGetAll(RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	private String getUriOfObject(String objectType, String objectUuid) {
+		Class<?> className;
+		OpenmrsObject openmrsObject = null;
+		try {
+			className = OpenmrsClassLoader.getInstance().loadClass(objectType);
+			openmrsObject = (OpenmrsObject) className.newInstance();
+		}
+		catch (ClassNotFoundException e) {}
+		catch (IllegalAccessException e) {}
+		catch (InstantiationException e) {
+			e.printStackTrace();
+		}
+		openmrsObject.setUuid(objectUuid);
+		String Uri = Context.getService(RestService.class)
+		        .getResourceByName("v1/" + objectType.substring(objectType.lastIndexOf(".") + 1).toLowerCase())
+		        .getUri(openmrsObject);
+		
```

Comment:
I think you're doing a little too much in this method, just fetch the rest resource by type after loading the class and call getUri()